catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/model/Search.java

```diff
@@ -172,6 +175,15 @@ private void updateStatus(String sourceId, QueryResponse queryResponse) {
                 State.SUCCEEDED :
                 State.FAILED));
         responseNum++;
+
+        fillReasonsFromStacktraceMessages(status, cause);
+    }
+
+    private void fillReasonsFromStacktraceMessages(QueryStatus status, Throwable nextCause) {
```

Comment:
I am not sure if we want to include all of the causes.  I think sending back the root message to the other system is good enough.